core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -371,7 +372,7 @@ public void moveBlock(long sessionId, long blockId, BlockStoreLocation oldLocati
         // Failed to create a temp block, so trigger Evictor to make some space.
         // NOTE: a successful {@link freeSpaceInternal} here does not ensure the subsequent
         // allocation also successful, because these two operations are not atomic.
-        freeSpaceInternal(sessionId, result.getBlockSize(), newLocation);
+        freeSpaceInternal(sessionId, result.getBlockSize(), newLocation, Mode.GUARANTEED);
```

Comment:
Are you planning to keep these calls to `freeSpaceInternal`?

Reply:
what do you mean? These calls are not affected by this PR, because those should be synchronous eviction.

Reply:
It is strange to have the guaranteed mode only accessible to internal methods. I guess your motivation was to avoid changing the interface?

Reply:
I am fine with keeping API unchanged since there seems no need to change the public API yet

Reply:
actually I saw the reserver is using this . Now I incline to maybe put mode in the public API to make it a bit cleaner.

@yupeng9  I am not strong on this. Up to you

Reply:
For now yes, I would consider changing it and adding the option to the method if we see other APIs of `freeSpace` in a guaranteed way outside this class.